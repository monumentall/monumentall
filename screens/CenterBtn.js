import React from "react";
import { View, Text } from "react-native";
import { specificStyles } from "../styles";

//Button to center user on map. Useful if they toggle their location back and forth, as map does not automatically re-center
const CenterBtn = ({ locationAccess, setMapRegion }) => {
  const buttonAction =
    locationAccess === "granted"
      ? () => setMapRegion("mapRegion")
      : () =>
          alert("Turn on location services to center map on your location.");
  return (
    <View>
      <Text style={specificStyles.centerBtn} onPress={buttonAction}>
        C
      </Text>
    </View>
  );
};

export default CenterBtn;
