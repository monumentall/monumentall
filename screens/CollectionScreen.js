import React from 'react';
import { ScrollView, StyleSheet, Text } from 'react-native';
import data from '../data';

export default class CollectionScreen extends React.Component {
  static navigationOptions = {
    title: 'Collection Details',
  };

  render() {
    return (
      <ScrollView style={styles.container}>
           <Text>Collection Details Page</Text>
           <Text>Landmark 1: {data[1].name}</Text>
      </ScrollView>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    paddingTop: 15,
    backgroundColor: '#fff',
  },
});
