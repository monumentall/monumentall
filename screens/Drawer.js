import React from "react";
import { View, Text } from "react-native";
import BottomDrawer from "rn-bottom-drawer";
import Landmark from "./Landmark";
import { reusableStyles } from "../styles";
import layout from "../constants/Layout";

const screenHeight = layout.window.height;

export default class Drawer extends React.Component {
  renderContent = () => {
    if (this.props.selectedLandmark.name) {
      return (
        <View>
          <Landmark landmarkDetails={this.props.selectedLandmark} />
        </View>
      );
    } else {
      return (
        <View>
          <Text style={reusableStyles.headline}>Explore Brooklyn</Text>
        </View>
      );
    }
  };

  render() {
    return (
      <BottomDrawer
        containerHeight={screenHeight - 25}
        startUp={false}
        roundedEdges={true}
        shadow={true}
        downDisplay={screenHeight - screenHeight / 6}
        backgroundColor={"#f2e5e5"}
      >
        {this.renderContent()}
      </BottomDrawer>
    );
  }
}
