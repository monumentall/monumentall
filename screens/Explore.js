import React from "react";
import PropTypes from 'prop-types';
import { connect } from "react-redux";
import { View, Text, TouchableOpacity } from "react-native";
import { specificStyles } from "../styles";
import Nearby from "./Nearby";
import Saved from "./Saved";
import Landmark from "./Landmark";
import BottomSheet from "reanimated-bottom-sheet";

class Explore extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      showSavedList: false,
      showNearby: true
    };
    this.bs = React.createRef();
  }

  showSavedListView = () => {
    this.setState({
      showNearby: false,
      showSavedList: true
    });
  };

  showNearbyView = () => {
    this.setState({
      showNearby: true,
      showSavedList: false
    });
  };

  componentDidUpdate = prevProps => {
    const propsChanged =
      prevProps.landmarkDetails !== this.props.landmarkDetails;
    const haveLandmark = !!this.props.landmarkDetails.name;

    if (propsChanged && haveLandmark) {
      this.setState({
        showNearby: false,
        showSavedList: false
      });
    }
  };

  displayHeader = () => {
    return (
      <View style={specificStyles.drawerButtonsBlock}>
        <TouchableOpacity onPress={this.showNearbyView}>
          <Text style={specificStyles.drawerButtons}>Nearby</Text>
        </TouchableOpacity>
        <TouchableOpacity onPress={this.showSavedListView}>
          <Text style={specificStyles.drawerButtons}>Saved</Text>
        </TouchableOpacity>
      </View>
    );
  };

  minimizeDrawer = () => {
    this.bs.current.snapTo(0);
  };

  displayContent = () => {
    const { showNearby, showSavedList } = this.state;
    const showLandmarkDetails =
      this.props.landmarkDetails.name && !showNearby && !showSavedList;
    return (
      <View>
        {showLandmarkDetails && (
          <Landmark
            landmarkDetails={this.props.landmarkDetails}
            closeDrawer={this.minimizeDrawer}
          />
        )}
        {showSavedList && <Saved />}
        {showNearby && <Nearby landmarks={this.props.landmarks} />}
      </View>
    );
  };

  render() {
    return (
      <BottomSheet
        ref={this.bs}
        snapPoints={[-500, 0]}
        initialSnap={0}
        renderContent={this.displayContent}
        renderHeader={this.displayHeader}
        enabledContentGestureInteraction={false}
      />
    );
  }
}

const mapStateToProps = state => ({
  landmarkDetails: state.selectedLandmark || {}
});

Explore.propTypes = {
  landmarkDetails: PropTypes.object.isRequired,
};

export default connect(mapStateToProps)(Explore);
