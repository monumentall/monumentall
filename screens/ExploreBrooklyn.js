import React from "react";
import { View, Text } from "react-native";
import { reusableStyles, specificStyles } from "../styles";

export default class ExploreBrooklyn extends React.Component {
  render() {
    const landmarks = this.props.landmarks;
    if (landmarks.length) {
      return (
        <View style={reusableStyles.block}>
          <Text style={{ ...reusableStyles.header1, textAlign: "center" }}>
            ExploreBrooklyn
          </Text>
          <View>
            {landmarks.map(landmark => (
              <View key={landmark.name} style={specificStyles.listItemWithIcon}>
                <View style={reusableStyles.listIcon} />
                <View>
                  <Text style={reusableStyles.header2}>{landmark.name}</Text>
                  <Text style={reusableStyles.text1}>{landmark.location}</Text>
                </View>
              </View>
            ))}
          </View>
        </View>
      );
    } else {
      return (
        <View>
          <Text>Explore Brooklyn</Text>
        </View>
      );
    }
  }
}
