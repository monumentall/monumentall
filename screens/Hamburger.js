import React from "react";
import { View } from "react-native";
import Hamburger from "react-native-hamburger";
import Menu from "./Menu.js";
import styles from "../styles";

export default class HamburgerModal extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      hamburgerActive: false
    };
  }

  render() {
    return (
      <View style={styles.hamburger}>
        <Hamburger
          active={this.state.hamburgerActive}
          type="cross"
          onPress={() =>
            this.setState({ hamburgerActive: !this.state.hamburgerActive })
          }
        />
        <Menu visible={this.state.hamburgerActive} />
      </View>
    );
  }
}
