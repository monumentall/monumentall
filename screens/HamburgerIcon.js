import React, { Component } from "react";
import { Animated, TouchableWithoutFeedback } from "react-native";

export default class HamburgerIcon extends Component {
  constructor(props) {
    super(props);
    this.state = {
      active: false
    };
  }

  cross() {
    if (!this.state.active) {
      Animated.spring(this.topBar, {
        toValue: 0.9
      }).start();
      Animated.spring(this.bottomBar, {
        toValue: 0.9
      }).start();
      Animated.spring(this.bottomBarMargin, {
        toValue: -10
      }).start();
      Animated.timing(this.middleBarOpacity, {
        toValue: 0,
        duration: 30
      }).start();
    } else {
      this.setState({
        active: false
      });
      Animated.spring(this.topBar, {
        toValue: 0
      }).start();
      Animated.spring(this.bottomBar, {
        toValue: 0
      }).start();
      Animated.spring(this.bottomBarMargin, {
        toValue: 4
      }).start();
      Animated.spring(this.middleBarOpacity, {
        toValue: 1,
        duration: 1200
      }).start();
    }
  }

  _animate() {
    setTimeout(() => {
      this.setState({
        active: this.props.active
      });
    }, 0);
    this.cross();
  }
  componentDidMount() {
    setTimeout(() => {
      this.setState({
        active: this.props.active
      });
    }, 0);
  }

  render() {
    const {
      props: { color }
    } = this;

    if (this.props.active) {
      this.topBar = this.topBar || new Animated.Value(0.9);
      this.bottomBar = this.bottomBar || new Animated.Value(0.9);
      this.bottomBarMargin = this.bottomBarMargin || new Animated.Value(-10);
      this.middleBarOpacity = this.middleBarOpacity || new Animated.Value(0);
    }

    this.containerAnim = this.containerAnim || new Animated.Value(0);
    this.topBar = this.topBar || new Animated.Value(0);
    this.bottomBar = this.bottomBar || new Animated.Value(0);
    this.middleBarOpacity = this.middleBarOpacity || new Animated.Value(1);
    this.bottomBarMargin = this.bottomBarMargin || new Animated.Value(4);
    this.topBarMargin = this.topBarMargin || new Animated.Value(0);
    this.marginLeft = this.marginLeft || new Animated.Value(0);
    this.width = this.width || new Animated.Value(25);

    return (
      <TouchableWithoutFeedback
        onPress={() => {
          this.props.onPress ? this.props.onPress() : undefined,
            this._animate();
        }}
      >
        <Animated.View
          style={{
            width: 35,
            justifyContent: "center",
            alignItems: "center",
            height: 35,
            transform: [
              {
                rotate: this.containerAnim.interpolate({
                  inputRange: [0, 1],
                  outputRange: ["0deg", "360deg"]
                })
              }
            ]
          }}
        >
          <Animated.View
            style={{
              height: 3,
              marginLeft: this.marginLeft,
              width: this.width,
              marginBottom: this.topBarMargin,
              backgroundColor: color ? color : "black",
              transform: [
                {
                  rotate: this.topBar.interpolate({
                    inputRange: [0, 1],
                    outputRange: ["0deg", "-50deg"]
                  })
                }
              ]
            }}
          />
          <Animated.View
            style={{
              height: 3,
              width: 25,
              opacity: this.middleBarOpacity,
              backgroundColor: color ? color : "black",
              marginTop: 4
            }}
          />
          <Animated.View
            style={{
              height: 3,
              marginLeft: this.marginLeft,
              width: this.width,
              backgroundColor: color ? color : "black",
              marginTop: this.bottomBarMargin,
              transform: [
                {
                  rotate: this.bottomBar.interpolate({
                    inputRange: [0, 1],
                    outputRange: ["0deg", "50deg"]
                  })
                }
              ]
            }}
          />
        </Animated.View>
      </TouchableWithoutFeedback>
    );
  }
}
