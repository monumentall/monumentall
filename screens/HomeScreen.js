import React from "react";
import PropTypes from 'prop-types';
import { connect } from "react-redux";
import { View, Text, TouchableOpacity } from "react-native";
import Map from "./Map";
import Explore from "./Explore";
import { specificStyles } from "../styles";
import { database } from "../db.js";
import { getLandmarksAction } from "../store/landmarks";

class HomeScreen extends React.Component {
  constructor(props) {
    super(props);
    this.db = database.ref();
  }

  componentDidMount = async () => {
    await this.props.getLandmarks(this.db);
  };

  render() {
    return !this.props.err ? (
      <View style={specificStyles.main}>
        <View>
          <Explore />
          <Map />
        </View>
      </View>
    ) : (
      <View style={specificStyles.main}>
        <Text>There was a problem loading the landmarks.</Text>
        <TouchableOpacity onClick={() => this.props.getLandmarks(this.db)}>
          <Text>Retry</Text>
        </TouchableOpacity>
      </View>
    );
  }
}

const mapStateToProps = state => ({
  landmarks: state.landmarks.data || [],
  err: state.landmarks.err
});

const mapDispatchToProps = dispatch => ({
  getLandmarks: db => dispatch(getLandmarksAction(db))
});

HomeScreen.propTypes = {
  err: PropTypes.bool.isRequired,
  getLandmarks: PropTypes.func.isRequired,
  landmarks: PropTypes.arrayOf(
    PropTypes.shape({
      coordinates: PropTypes.objectOf(
        PropTypes.shape({
          latitude: PropTypes.number.isRequired,
          latitudeDelta: PropTypes.number.isRequired,
          longitude: PropTypes.number.isRequired,
          longitudeDelta: PropTypes.number.isRequired,
        }),
      ),
      description: PropTypes.string.isRequired,
      image: PropTypes.string.isRequired,
      location: PropTypes.string.isRequired,
      name: PropTypes.string.isRequired,
    }),
  ),
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(HomeScreen);
