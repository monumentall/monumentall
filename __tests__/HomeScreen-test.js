import React from "react";
import { Text } from "react-native";
import { configure, shallow } from "enzyme";
import { HomeScreen } from "../screens/HomeScreen";
import Adapter from "enzyme-adapter-react-16";

configure({
  adapter: new Adapter()
});

const baseProps = {
  err: false,
  landmarks: [],
  getLandmarks: () => []
};

describe("Home Screen", () => {
  it("should render error text if there is an error", () => {
    const props = { ...baseProps, err: true };

    const wrapper = shallow(<HomeScreen {...props} />);
    expect(wrapper.find(Text)).toHaveLength(2);
    expect(
      wrapper
        .find(Text)
        .first()
        .html()
    ).toEqual("<Text>There was a problem loading the landmarks.</Text>");
  });
});
