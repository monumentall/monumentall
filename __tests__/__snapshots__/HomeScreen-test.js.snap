// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeScreen renders the map and the "Explore Brooklyn" drawer as the default 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View>
    <MapView
      initialRegion={null}
      onRegionChange={[Function]}
      region={
        Object {
          "latitude": 40.673868,
          "latitudeDelta": 0.0922,
          "longitude": -73.970089,
          "longitudeDelta": 0.0421,
        }
      }
      showsUserLocation={true}
      style={
        Object {
          "height": 1334,
          "width": 750,
        }
      }
    >
      <View>
        <Text
          onPress={[Function]}
          style={
            Object {
              "fontSize": 40,
              "marginLeft": 15,
              "marginTop": 25,
              "width": 40,
            }
          }
        >
          X
        </Text>
        <View />
      </View>
      <View>
        <Text
          onPress={[Function]}
          style={
            Object {
              "fontSize": 40,
              "marginLeft": 250,
              "marginTop": 25,
              "width": 40,
            }
          }
        >
          C
        </Text>
      </View>
    </MapView>
    <View
      onMoveShouldSetResponder={[Function]}
      onMoveShouldSetResponderCapture={[Function]}
      onResponderEnd={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderReject={[Function]}
      onResponderRelease={[Function]}
      onResponderStart={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      onStartShouldSetResponderCapture={[Function]}
      style={
        Object {
          "backgroundColor": "#f2e5e5",
          "borderTopLeftRadius": 10,
          "borderTopRightRadius": 10,
          "height": 1345.5239647354992,
          "left": 0,
          "position": "absolute",
          "shadowColor": "#CECDCD",
          "shadowOpacity": 5,
          "shadowRadius": 3,
          "top": 1136.6666666666667,
          "width": 750,
        }
      }
    >
      <View>
        <Text
          style={
            Object {
              "alignSelf": "center",
              "marginTop": 25,
            }
          }
        >
          Explore Brooklyn
        </Text>
      </View>
      <View
        style={
          Object {
            "backgroundColor": "#f2e5e5",
            "height": 36.52396473549935,
          }
        }
      />
    </View>
  </View>
</View>
`;
